lib/Model.ts

```diff
@@ -692,6 +693,52 @@ export class Model<TDocument extends { _id?: any }, TInstance> {
         });
     }
 
+    private _mapReduce<Key, Value>(instanceType: InstanceImplementation<MapReducedDocument<Key, Value>, any>, options: MongoDB.MapReduceOptions): Bluebird<void> {
```

Comment:
Looking at the implementations, you may be able to write things such that you don't need to branch calls to `this.collection.mapReduce()` (by extracting the `mapReduceOptions` first and checking whether `options.out` is `{ inline: 1 }` in the callback before `resolve(data)`).